### Eclipse Workspace Patch 1.0
#P Calendar
Index: src/com/vaadin/addon/calendar/gwt/client/ui/schedule/WeekGrid.java
===================================================================
--- src/com/vaadin/addon/calendar/gwt/client/ui/schedule/WeekGrid.java	(revision 15690)
+++ src/com/vaadin/addon/calendar/gwt/client/ui/schedule/WeekGrid.java	(working copy)
@@ -1462,6 +1462,7 @@
                 MouseDownHandler, MouseUpHandler, MouseMoveHandler,
                 KeyDownHandler {
 
+            private static Element currentlySelectedElement;
             private Element caption = null;
             private final Element eventContent;
             private CalendarEvent calendarEvent = null;
@@ -1723,6 +1724,13 @@
                                         calendar.getPID(),
                                         CalendarEventId.EVENTCLICK,
                                         calendarEvent.getIndex(), true);
+                                
+                                if (currentlySelectedElement != null) {
+                                    currentlySelectedElement.removeClassName("selected");
+                                }
+                                getElement().addClassName("selected");
+                                currentlySelectedElement = getElement();
+                                
                             }
                         }
                     }
